hypervisor/arch/x86/configs/vm_config.c

```diff
@@ -163,27 +152,23 @@ bool sanitize_vm_config(void)
 			} else if (is_safety_vm_uuid(vm_config->uuid) && (vm_config->severity != (uint8_t)SEVERITY_SAFETY_VM)) {
 				ret = false;
 			} else {
-				pre_launch_pcpu_bitmap |= vm_pcpu_bitmap;
+				pre_launch_pcpu_bitmap |= vm_config->pcpu_affinity_bitmap;
 			}
 			break;
 		case SOS_VM:
 			/* Deduct pcpus of PRE_LAUNCHED_VMs */
-			sos_pcpu_bitmap ^= pre_launch_pcpu_bitmap;
-			if ((sos_pcpu_bitmap == 0U) || ((vm_config->guest_flags & GUEST_FLAG_LAPIC_PASSTHROUGH) != 0U)) {
+			vm_config->pcpu_affinity_bitmap = ((1UL << get_pcpu_nums()) - 1UL) ^ pre_launch_pcpu_bitmap;
+			vm_config->vcpu_num = bitmap_weight(vm_config->pcpu_affinity_bitmap);
+			if ((vm_config->pcpu_affinity_bitmap == 0U) ||
```

Comment:
using vm_config->vcpu_num == 0 to replace bitmap == 0? Seems vcpu_num is more readable.

Reply:
Yes, agreed.